services/worker-manager/test/api_test.js

```diff
@@ -727,6 +727,48 @@ helper.secrets.mockSuite(testing.suiteName(), [], function(mock, skipping) {
       assert.equal(worker.expires, expires.toJSON());
     });
 
+    test('update a worker for a worker pool that does not exist', async function () {
+      await assert.rejects(() =>
+        helper.workerManager.updateWorker(workerPoolId, workerGroup, workerId, {
+          expires: taskcluster.fromNow('1 hour'),
+          capacity: 1,
+        }), new RegExp(`Worker pool ${workerPoolId} does not exist`));
+    });
+
+    test('update a worker for a worker pool with invalid providerId', async function () {
+      await createWorkerPool({ providerId: 'nosuch' });
+      await assert.rejects(() =>
+        helper.workerManager.updateWorker(workerPoolId, workerGroup, workerId, {
+          expires: taskcluster.fromNow('1 hour'),
+          capacity: 1,
+        }), /Provider nosuch for worker pool/);
+    });
+
+    test('update a worker for a provider that does not want it', async function () {
+      await createWorkerPool({});
+      await createWorker({});
+      await assert.rejects(() =>
+        helper.workerManager.updateWorker(workerPoolId, workerGroup, workerId, {
+          expires: taskcluster.fromNow('1 hour'),
+          capacity: 1,
+        }), /updating workers is not supported/);
+    });
+
+    test('update a worker for a provider that wants and appreciates it', async function () {
+      await createWorkerPool({
+        providerData: { allowUpdateWorker: true },
+      });
+      await createWorker({});
+      const worker = await helper.workerManager.updateWorker(
+        workerPoolId,
+        workerGroup,
+        workerId, {
+          expires: taskcluster.fromNow('1 hour'),
+          capacity: 2,
+        });
+      assert.deepEqual(worker.capacity, 2);
+    });
```

Comment:
Should we test any race conditions here, such as a parallel worker termination and worker update?

Reply:
I'm not sure what you have in mind here.  What's the failure case?

Reply:
I haven't really understood what the `updateWorker` call is for. I thought it was about refreshing the config on a running worker, but above you state that it isn't that. My suggestion was based on this misunderstanding, so it may not apply.